File: src/Services/EcommerceDDD.Quotes/API/Controllers/Requests/AddQuoteItemRequest.cs
using System.ComponentModel.DataAnnotations;

namespace EcommerceDDD.Quotes.API.Controllers.Requests;

public record class AddQuoteItemRequest
{
    [Required(ErrorMessage = "The {0} field is required.")]
    public Guid ProductId { get; init; }

    [Required(ErrorMessage = "The {0} field is required.")]
    public int Quantity { get; init; }

    [Required(ErrorMessage = "The {0} field is required.")]
    public string CurrencyCode { get; init; }
}

File: src/EcommerceDDD.Spa/src/app/core/models/requests/AddQuoteItemRequest.ts
export class AddQuoteItemRequest {
  productId: string;
  quantity: number;
  currencyCode: string;

  constructor(productId: string, quantity: number, currencyCode: string) {
    this.productId = productId;
    this.quantity = quantity;
    this.currencyCode = currencyCode;
  }
}
